wrappers/desktop.go

```diff
@@ -232,7 +232,7 @@ func AddSnapDesktopFiles(s *snap.Info) (err error) {
 
 // RemoveSnapDesktopFiles removes the added desktop files for the applications in the snap.
 func RemoveSnapDesktopFiles(s *snap.Info) error {
-	glob := filepath.Join(dirs.SnapDesktopFilesDir, s.Name()+"_*.desktop")
+	glob := filepath.Join(dirs.SnapDesktopFilesDir, s.InstanceName()+"_*.desktop")
```

Comment:
Hmm, can you add a TODO about inspecting the instance-safety of this code.
I'm surprised by the use of `_` in the glob.